api/src/lib/connector.js

```diff
@@ -79,7 +79,10 @@ module.exports = class Conncetor {
     if (peerInfo.publicKey) return peerInfo
 
     // Get the host publicKey
-    const hostInfo = yield this.utils.hostLookup('https://' + peer.hostname)
+    const secure = this.config.getIn(['server', 'public_secure'])
+    const isLocalhost = peer.hostname.trim().toLowerCase().startsWith('localhost')
```

Comment:
This would cause potential issues on domains that start with `localhost` like `localhost1.com`.